From 18d2e8eaf68c8b4dd53b1f27bd4add745765b19e Mon Sep 17 00:00:00 2001
From: Aleksey Vasenev <margtu-fivt@ya.ru>
Date: Thu, 28 Jul 2016 15:31:06 +0300
Subject: [PATCH] Align thread entry point stack

__attribute__((aligned)) don't work for stack variables in threads created
with _beginthreadex without alignment.

Signed-off-by: Aleksey Vasenev <margtu-fivt@ya.ru>
---
 mingw-w64-libraries/winpthreads/src/thread.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mingw-w64-libraries/winpthreads/src/thread.c b/mingw-w64-libraries/winpthreads/src/thread.c
index ceea98e..9fa773a 100644
--- a/mingw-w64-libraries/winpthreads/src/thread.c
+++ b/mingw-w64-libraries/winpthreads/src/thread.c
@@ -1454,6 +1454,9 @@ pthread_setcanceltype (int type, int *oldtype)
   return 0;
 }
 
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2)
+__attribute__((force_align_arg_pointer))
+#endif
 int
 pthread_create_wrapper (void *args)
 {
-- 
2.9.1

